publication-commons/src/main/java/no/unit/nva/doi/models/Doi.java

```diff
@@ -0,0 +1,75 @@
+package no.unit.nva.doi.models;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+
+/**
+ * Doi class for working with Dois.
+ *
+ * <p>Use {@link Doi#builder()} for constructing a new Doi instance.
+ */
+public abstract class Doi {
+
+    public static final String HANDLE_DOI_PREFIX = "10.";
+    public static final String ERROR_PROXY_URI_MUST_BE_A_VALID_URL = "Proxy URI must be a valid URL";
+    protected static final String DOI_ORG = "doi.org";
+    protected static final String HANDLE_STAGE_DATACITE_ORG = "handle.stage.datacite.org";
+    protected static final String DX_DOI_ORG = "dx.doi.org";
+    public static final List<String> VALID_PROXIES = List.of(DOI_ORG, DX_DOI_ORG, HANDLE_STAGE_DATACITE_ORG);
+    protected static final String HTTPS = "https";
+    protected static final String HTTP = "http";
+    public static final List<String> VALID_SCHEMES = List.of(HTTPS, HTTP);
```

Comment:
What use case do we have that requires a DOI to be anything else than a URL type object that can be validated?

Reply:
@janbj : FYI, https://github.com/BIBSYSDEV/nva-publication-api/pull/96#issuecomment-730971822 

URL#equals etc does DNS lookups etc which is not nice. Its a long term common thing to avoid URL ( https://bugs.java.com/bugdatabase/view_bug.do?bug_id=4434494 ) . It has sadly other annoying things as well when it comes to percent encoding etc.